### DISCUSSION TITLE
Personal API keys

### DISCUSSION THREAD

**Discussion by korridor** [solidtime-io/solidtime#274] [Feature requests]
Add ability to create personal API keys for in the user interface. This should make it possible to write custom integrations. There is aready an [API documentation](https://docs.solidtime.io/api-reference).

**Comment by cupcakearmy** [comment]
I was searching for this, as I would like to create a Raycast extension for Solidtime. Love the product.

**Comment by henningtillmann** [comment]
API Keys would be great. I'd like to create an iOS / macOS Shortcut. So I can start a timer directly from my Apple Watch :-).
I will share the shortcut as soon as I figure out how to properly authenticate.

**Reply by korridor** [reply]
Thanks great to hear that people want to integrate into solidtime. API keys are the next thing, I'll work on and it looks like it's going to be ready soon!

**Comment by korridor** [comment]
This feature is currently a work-in-progress. You can follow the progress in the following draft PR: #366

**Comment by korridor** [comment]
This feature is now released in solidtime Cloud and is pushed in `main`. We'll add docs for how to use that for self-hosting in the next days. Until then you can try it in the solidtime Cloud version.

**Comment by korridor** [comment]
This feature is now released in version [0.6.0](https://github.com/solidtime-io/solidtime/releases/tag/v0.6.0).

Thanks for your comments: @cupcakearmy @henningtillmann

**Reply by henningtillmann** [reply]
When I try to create an API key I get the following message: "Personal access client is not configured". What does that mean?

**Reply by korridor** [reply]
@henningtillmann You need to do the "After update tasks" in the [release note of version 0.6.0](https://github.com/solidtime-io/solidtime/releases/tag/v0.6.0) to use this feature.

In there it says to create a personal access client. Here is the direct link to the [step in the self-hosting guide](https://github.com/solidtime-io/solidtime/releases/tag/v0.6.0).

**Reply by henningtillmann** [reply]
Oh damn, I haven't seen it. Sorry! Thank you!

**Reply by cupcakearmy** [reply]
https://github.com/raycast/extensions/pull/17234 Created a raycast extension :)

**Reply by korridor** [reply]
@henningtillmann No worries :)

**Comment by henningtillmann** [comment]
I have created an [iOS/watchOS/macOS Shortcut](https://github.com/solidtime-io/solidtime/discussions/389)
